### PR TITLE
Adding Linux build to Travis CI

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [ ] New extensions are written in Swift 4.
- [ ] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [ ] I have added tests for new extensions, and they passed.
- [ ] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [ ] All extensions are declared as **public**.
- [ ] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,36 +2,44 @@ os: osx
...
+        - xcodebuild clean build test -project "$PROJECT" -scheme "$TVOS_SCHEME" -destination "$TVOS_DESTINATION" | XCPRETTY_JSON_FILE_OUTPUT="xcodebuild-tvos.json" xcpretty -f `xcpretty-json-formatter`
+        - bash <(curl -s https://codecov.io/bash) -cF tvos -J 'SwifterSwift'
+        - xcodebuild clean build test -project "$PROJECT" -scheme "$MACOS_SCHEME" -destination "$MACOS_DESTINATION" | XCPRETTY_JSON_FILE_OUTPUT="xcodebuild-macos.json" xcpretty -f `xcpretty-json-formatter`
+        - bash <(curl -s https://codecov.io/bash) -cF osx -J 'SwifterSwift'
+        - xcodebuild clean build -project "$PROJECT" -scheme "$WATCHOS_SCHEME" -destination "$WATCHOS_DESTINATION" | XCPRETTY_JSON_FILE_OUTPUT="xcodebuild-watchos.json" xcpretty -f `xcpretty-json-formatter`
+        - bundle exec danger --verbose
+        - bundle exec pod lib lint --skip-tests --allow-warnings
+    - name: Linux 
+      env: SWIFT_VERSION=5.0
+      os: linux
```
Comment (guykogus) [inline .travis.yml]: Does this mean that the tests only run on Linux?

**Comment by omaralbeik** [comment]
@LucianoPAlmeida While the project is compiling with no problem on Linux, tests are another story 😄, I think we need to spend more time on this

File: .travis.yml (hunk 1 above)
Comment (LucianoPAlmeida) [reply .travis.yml]: nope, they run on both Darwin and Linux :)

**Comment by LucianoPAlmeida** [comment]
@guykogus I think we can do this 👍

**Comment by LucianoPAlmeida** [comment]
Guys, we are facing some segfaults on some date extensions because of some issues on core-libs-foundation. 
See [SR-10147](https://bugs.swift.org/browse/SR-10147) and [SR-9101](https://bugs.swift.org/browse/SR-9101) 
We should probably stale this for while or disable those extensions on Linux for now. 
cc @omaralbeik

**Comment by omaralbeik** [comment]
Thank you @LucianoPAlmeida for taking care of this, awesome work!

**Pull request merged**